editor/src/java/com/defold/libs/ResourceUnpacker.java

```diff
@@ -59,13 +62,19 @@ private static void deleteOldUnpackDirs(Path unpackPath) throws IOException {
         logger.info("deleting old unpack dirs from {}", unpackRoot);
         final long now = new Date().getTime();
         final long oneWeekAgo = now - (7 * 24 * 60 * 60 * 1000);
-        for (File unpackDir : unpackRoot.toFile().listFiles(File::isDirectory)) {
-            long creationTime = Files.getLastModifiedTime(unpackDir.toPath()).toMillis();
-            if (creationTime < oneWeekAgo) {
-                logger.info("deleting unpack dir {}", unpackDir);
-                FileUtils.deleteQuietly(unpackDir);
-            }
-        }
+        Files.list(unpackRoot)
+                .filter(path -> Files.isDirectory(path) && !path.equals(unpackPath))
+                .forEach(unpackDir -> {
+                    try {
+                        long creationTime = Files.getLastModifiedTime(unpackDir).toMillis();
+                        if (creationTime < oneWeekAgo) {
+                            logger.info("deleting unpack dir {}", unpackDir);
+                            FileUtils.deleteQuietly(unpackDir.toFile());
+                        }
+                    } catch (IOException e) {
+                        logger.warn("Couldn't get lastModifiedTime of {}", unpackDir, e);
+                    }
+                });
```

Comment:
mostly a refactoring from Files to Paths API, though there is one useful addition — we no longer delete the unpack path itself.